include/pros/rtos.hpp

```diff
@@ -357,7 +357,7 @@ class Mutex {
 	bool give(void);
 
 	private:
-	mutex_t mutex;
+	std::shared_ptr<std::remove_pointer_t<mutex_t>> mutex;
```

Comment:
I don't think you need this remove pointer as well.

Reply:
What he's doing is avoiding an additional layer of heap allocation and indirection, by using the fact that `mutex_t` is a pointer. I support the way he's currently implemented it.